tools/run_tests/performance/loadtest_config.py

```diff
@@ -118,17 +118,54 @@ def gen_run_indices(runs_per_test: int) -> Iterable[str]:
         yield index_fmt.format(i)
 
 
+def scenario_transform_function(
+    client_channels: int, offered_loads: Iterable[int],
+    async_server_threads: int
+) -> Optional[Callable[[Iterable[Mapping[str, Any]]], Iterable[Mapping[str,
+                                                                       Any]]]]:
+    """Returns a transform to be applied to a list of scenarios."""
+    if client_channels == 0 or len(
+            offered_loads) == 0 or async_server_threads == 0:
+        return lambda s: s
+
+    def _transform(
+            scenarios: Iterable[Mapping[str,
+                                        Any]]) -> Iterable[Mapping[str, Any]]:
+        """Transforms scenarios by inserting num of client channels, number of async_server_threads and offered_load."""
+
+        for base_scenario in scenarios:
+            base_scenario['client_config']['client_channels'] = client_channels
+            base_scenario['server_config'][
+                'async_server_threads'] = async_server_threads
+            base_name = base_scenario['name']
+            for offered_load in offered_loads:
```

Comment:
There may be a wrinkle here if we assume that len(offered_loads) may be zero (and in that case we just return the scenario without changing the offered load). I would handle that by having a function scenario_name for the name:

```python
def scenario_name(base_name, client_channels, server_threads, offered_load):
  elements = [ base_name ]
  if client_channels:
    elements.append(f'{%d}channels', client_channels)
  if server_threads:
    elements.append(f'{%d}threads', server_threads)
  if offered_load:
    elements.append(f{%d}load', offered_load)
  return '_´.join(elements)
```

Given this function, I would then write the loop as follows:

```python
base_name = base_scenario['name']

if not offered_loads:
  base_scenario['name'] = scenario_name(base_name, client_channels, server_threads, 0)
  yield base_scenario
  return

for offered_load in offered_loads:
  scenario = copy.deepcopy(base_scenario)
  scenario['name'] = scenario_name(base_name, client_channels, server_threads, offered_load)
  scenario['client_config´]['load_params'] = { 'poisson': { 'offered_load': offered_load } }
```